disnake/ui/action_row.py

```diff
@@ -264,19 +353,141 @@ def add_text_input(
                 required=required,
                 min_length=min_length,
                 max_length=max_length,
-            )
+            ),
+        )
+
+    def clear_items(self) -> None:
+        """Remove all components from the action row.
+
+        .. versionadded:: 2.6
+        """
+        self._children.clear()
+
+    def remove_item(self, item: UIComponentT) -> None:
+        """Remove a component from the action row.
+
+        .. versionadded:: 2.6
+
+        Parameters
+        ----------
+        item: :class:`WrappedComponent`
+            The component to remove from the action row.
+
+        Raises
+        ------
+        ValueError:
+            The component could not be found on the action row.
+        """
+        self._children.remove(item)
+
+    def pop(self, index: int) -> UIComponentT:
+        """Pop the component at the provided index from the action row.
+        This takes into account the width of components.
+
+        .. versionadded:: 2.6
+
+        Parameters
+        ----------
+        index: :class:`int`
+            The index at which to pop the component.
+
+        Raises
+        ------
+        IndexError
+            There is no component at the provided index.
+        """
+        self.remove_item(component := self[index])
+        return component
+
+    @property
+    def _underlying(self) -> ActionRowComponent:
+        return ActionRowComponent._raw_construct(
+            type=self.type,
+            children=[comp._underlying for comp in self._children],
         )
 
     def to_component_dict(self) -> ActionRowPayload:
         return self._underlying.to_dict()
 
+    def __delitem__(self, index: int) -> None:
+        del self._children[index]
```

Comment:
unlike `__getitem__`, this doesn't take component width into account.
see the main review comment as well though

Reply:
Mhm, I can agree with the main review comment. I've removed the component width checking for indexing altogether.